src/drivers/intel/cavs/ipc.c

```diff
@@ -128,12 +128,54 @@ static void ipc_irq_handler(void *arg)
 	}
 }
 
+#if CAVS_VERSION >= CAVS_VERSION_1_8
+int ipc_platform_compact_read_msg(ipc_cmd_hdr *hdr, int words)
+{
+	uint32_t *chdr = (uint32_t *)hdr;
+
+	/* compact messages are 2 words on CAVS 1.8 onwards */
+	if (words != 2)
+		return 0;
+
+	chdr[0] = ipc_read(IPC_DIPCTDR);
+	chdr[1] = ipc_read(IPC_DIPCTDD);
+
+	return 2; /* number of words read */
+}
+
+int ipc_platform_compact_write_msg(ipc_cmd_hdr *hdr, int words)
+{
+	uint32_t *chdr = (uint32_t *)hdr;
+
+	/* compact messages are 2 words on CAVS 1.8 onwards */
+	if (words != 2)
+		return 0;
+
+	/* command complete will set the busy/done bits */
+	ipc_write(IPC_DIPCTDR, chdr[0] & ~IPC_DIPCTDR_BUSY);
+	ipc_write(IPC_DIPCTDD, chdr[1]);
+
+	return 2; /* number of words written */
```

Comment:
@lrgirdwo This function is not fit for IPC4 which send message to host by dsp ipc registers IPC_DIPCIDR & IPC_DIPCIDD.  The message may be a ack of host ipc messge,  fw notification or other fw message. And we can't set above registers directly to avoid fw ipc message conflict.  We have to go normal way to send it or queue it.  I will update it in IPC4 branch

Reply:
I need to comment this more, will update it an a subsequent PR, but this is just to populate the compact message and not send it.